pkg/controller/sync/resource.go

```diff
@@ -80,27 +77,21 @@ func (r *federatedResource) TargetName() util.QualifiedName {
 }
 
 func (r *federatedResource) Object() *unstructured.Unstructured {
-	if r.targetIsNamespace {
-		return r.placement
-	}
-	return r.template
+	return r.federatedResource
 }
 
 func (r *federatedResource) TemplateVersion() (string, error) {
-	return GetTemplateHash(r.template)
+	obj := r.federatedResource
+	if r.targetIsNamespace {
+		obj = r.namespace
+	}
+	return GetTemplateHash(obj.Object, r.targetIsNamespace)
```

Comment:
[Query]: Apologies for asking as this question is not from the change in this PR. I probably missed it when version manager was introduced. Why is a hash calculated and used in version manager for federated resources? Do CRD resources not get their own ResourceVersion` values auto updated on changes? Or this is to be able to determine part of the whole object (say `spec.overrides` now)?

Reply:
Resource version changes when anything in the resource changes, so it's not appropriate for determining the identity of a field (e.g. template, overrides).  The switch to using a hash for the template was made in https://github.com/kubernetes-sigs/federation-v2/pull/527, and this PR switches to using a hash for overrides too since it is no longer in its own resource.

Reply:
Got it, thanks!